pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFormatSpec.java

```diff
@@ -36,7 +37,9 @@
 public class DateTimeFormatSpec {
 
   public static final String NUMBER_REGEX = "[1-9][0-9]*";
+  public static final String COLON_REGEX = "([0-9])(:)(\\w+)(:)(\\w+)(:?)(.+)";
```

Comment:
Can we name this variable to a better one. Also, add a brief comment on why this is needed (mention the backward compatibility)

Reply:
Removed this regex matcher to check if the first character of the format is a digit or not.